src/store/User/userUtils.ts

```diff
@@ -22,17 +26,12 @@ export const recoverUser = async (
   try {
     const payload = await setupUser(
       keystore,
-      (payload: { contact: Contact; message: Message }) => {
-        dispatch({
-          type: USER_ACTIONS.RECEIVE_MESSAGE,
-          payload,
-        })
-      },
+      receiveMessage(dispatch)
     )
     dispatch({ type: USER_ACTIONS.SET_CLIENT, payload })
     const newContacts = contacts
       ? await Promise.all(
-        contacts.map((contact: IContactParams) => Contact.new(contact)),
+        contacts.map(async (contact: IContactParams) => await Contact.new(contact)),
```

Comment:
this is unnecessary, you have it in `Promise.all`

Reply:
true